pkg/api/validation/validation.go

```diff
@@ -2272,6 +2276,12 @@ func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateHostAliases(spec.HostAliases, fldPath.Child("hostAliases"))...)
 	}
 
+	if len(spec.PriorityClassName) > 0 {
```

Comment:
Makes sense. I added a check for preventing `Priority` from being set if the feature is not enabled.